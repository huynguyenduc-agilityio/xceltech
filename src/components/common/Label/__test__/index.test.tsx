import { render, screen } from '@testing-library/react';
import { Label, labelVariants } from '..';

describe('Label Component', () => {
  it('renders with default props', () => {
    render(<Label data-testid="test-label">Test Label</Label>);

    const label = screen.getByTestId('test-label');

    expect(label).toHaveClass(
      'peer-disabled:cursor-not-allowed peer-disabled:opacity-70',
    );
  });

  it('applies custom className', () => {
    render(
      <Label data-testid="custom-label" className="custom-class">
        Custom Label
      </Label>,
    );

    const label = screen.getByTestId('custom-label');

    expect(label).toHaveClass('custom-class');
  });

  it('generates correct class names with labelVariants', () => {
    const defaultClasses = labelVariants();

    expect(defaultClasses).toContain('text-black-soft');
    expect(defaultClasses).toContain('text-xl');
  });

  it('should match snapshot', () => {
    const { container } = render(<Label>Test Label</Label>);

    expect(container).toMatchSnapshot();
  });
});
