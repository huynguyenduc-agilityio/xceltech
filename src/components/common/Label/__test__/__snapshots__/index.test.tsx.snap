// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component should match snapshot 1`] = `
<div>
  <label
    class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 text-black-soft text-xl"
  >
    Test Label
  </label>
</div>
`;
